sinks/sinks_service.go

```diff
@@ -62,71 +65,191 @@ func (svc sinkService) CreateSink(ctx context.Context, token string, sink Sink)
 	sink.ID = id
 
 	// After creating, decrypt Metadata to send correct information to Redis
-	sink, err = svc.decryptMetadata(sink)
-
+	sink, err = svc.decryptMetadata(cfg, sink)
+	if err != nil {
+		return Sink{}, errors.Wrap(ErrCreateSink, err)
+	}
 	return sink, nil
 }
 
-func (svc sinkService) encryptMetadata(sink Sink) (Sink, error) {
-	var err error
-	sink.Config.FilterMap(func(key string) bool {
-		return key == backend.ConfigFeatureTypePassword
-	}, func(key string, value interface{}) (string, interface{}) {
-		var stringVal string
-		switch v := value.(type) {
-		case *string:
-			stringVal = *v
-		case string:
-			stringVal = v
+func validateAuthType(s *Sink) (authentication_type.AuthenticationType, error) {
+	var authMetadata types.Metadata
+	if len(s.ConfigData) != 0 {
+		var helper types.Metadata
+		if s.Format == "yaml" {
+			err := yaml.Unmarshal([]byte(s.ConfigData), &helper)
+			if err != nil {
+				return nil, err
+			}
+			authMetadata = helper.GetSubMetadata("authentication")
```

Comment:
I would have liked to see this "authentication" string as a constant, rather than typing the string everywhere which is prone to typos